server/lib/PasswordUtils.cc

```diff
@@ -0,0 +1,26 @@
+#include <botan/argon2.h>
+#include <botan/system_rng.h>
+
+#include "PasswordUtils.hh"
+
+namespace server {
+
+std::string
+PasswordUtils::hashPassword(std::string const& password)
+{
+    return Botan::argon2_generate_pwhash(
+      password.c_str(),
+      password.size(),
+      Botan::system_rng(),
+      1,
+      65536, // TODO(papaj-na-wrotkach): move to config
+      1);
+}
+
+bool
+PasswordUtils::checkHash(std::string const& hash, std::string const& password)
+{
+    return Botan::argon2_check_pwhash(password.c_str(), password.size(), hash);
+}
```

Comment:
Botan takes `char const*`, not an `std::string`, so it'd be better to take `std::string_view`s as arguments. By doing this we avoid a situation where we only have a `char const*`, and have to unnecessarily convert it to `std::string`.

And yes, `std::string_view` has a size function.

Reply:
To elaborate,
`std::string_view` does not make any dynamic memory allocations, but only holds a pointer to the begging, and size of the string. That's why it's preferred for read-only string 'views'.